x-pack/plugins/security_solution/common/endpoint/types/index.ts

```diff
@@ -251,152 +334,133 @@ export interface Host {
 /**
  * A record of hashes for something. Provides hashes in multiple formats. A favorite structure of the Elastic Endpoint.
  */
-interface Hashes {
+type Hashes = Partial<{
   /**
    * A hash in MD5 format.
    */
-  md5: string;
+  md5: ECSField<string>;
   /**
    * A hash in SHA-1 format.
    */
-  sha1: string;
+  sha1: ECSField<string>;
   /**
    * A hash in SHA-256 format.
    */
-  sha256: string;
-}
+  sha256: ECSField<string>;
+}>;
 
-interface MalwareClassification {
-  identifier: string;
-  score: number;
-  threshold: number;
-  version: string;
-}
+type MalwareClassification = Partial<{
+  identifier: ECSField<string>;
+  score: ECSField<number>;
+  threshold: ECSField<number>;
+  version: ECSField<string>;
+}>;
 
-interface ThreadFields {
-  id: number;
-  Ext: {
-    service_name: string;
-    start: number;
-    start_address: number;
-    start_address_module: string;
-  };
-}
+type ThreadFields = Partial<{
+  id: ECSField<number>;
+  Ext: Partial<{
+    service_name: ECSField<string>;
+    start: ECSField<number>;
+    start_address: ECSField<number>;
+    start_address_module: ECSField<string>;
+  }>;
+}>;
 
-interface DllFields {
+type DllFields = Partial<{
   hash: Hashes;
-  path: string;
-  pe: {
-    architecture: string;
-  };
-  code_signature: {
-    subject_name: string;
-    trusted: boolean;
-  };
-  Ext: {
-    compile_time: number;
+  path: ECSField<string>;
+  pe: Partial<{
+    architecture: ECSField<string>;
+  }>;
+  code_signature: Partial<{
+    subject_name: ECSField<string>;
+    trusted: ECSField<boolean>;
+  }>;
+  Ext: Partial<{
+    compile_time: ECSField<number>;
     malware_classification: MalwareClassification;
-    mapped_address: number;
-    mapped_size: number;
-  };
-}
+    mapped_address: ECSField<number>;
+    mapped_size: ECSField<number>;
+  }>;
+}>;
 
 /**
  * Describes an Alert Event.
  */
-export interface AlertEvent {
-  '@timestamp': number;
-  agent: {
-    id: string;
-    version: string;
-    type: string;
-  };
-  ecs: {
-    version: string;
-  };
-  event: {
-    id: string;
-    action: string;
-    category: string;
-    kind: string;
-    dataset: string;
-    module: string;
-    type: string;
-    sequence: number;
-  };
-  Endpoint: {
-    policy: {
-      applied: {
-        id: string;
-        status: HostPolicyResponseActionStatus;
-        name: string;
-      };
-    };
-  };
-  process: {
-    command_line?: string;
-    pid: number;
-    ppid?: number;
-    entity_id: string;
-    parent?: {
-      pid: number;
-      entity_id: string;
-    };
-    name: string;
-    hash: Hashes;
-    executable: string;
-    start: number;
-    thread?: ThreadFields[];
+export type AlertEvent = Partial<{
```

Comment:
Wrapping everything in `ECSField` and removing the duplication between this type and `SafeEndpointEvent` since we'll just combine them